packages/aws-cdk/bin/cdk.ts

```diff
@@ -264,8 +264,8 @@ async function initCommandLine() {
             environmentGlobs = [ '**' ]; // default to ALL
         }
         const stackInfos = await selectStacks();
-        const availableEnvironments = stackInfos.map(stack => stack.environment)
-                                                .filter(env => env !== undefined);
+        const availableEnvironments = distinct(stackInfos.map(stack => stack.environment)
+                                                         .filter(env => env !== undefined) as cxapi.Environment[]);
```

Comment:
Typecast should not be necessary, from the type of `distinct()` ?

Reply:
The type that `tsc` infers is an array of optionnals, as it cannot push out the `.filter` restriction to non-`undefined`, so I have to help it a little :)

Reply:
Ah yeah